src/main/java/net/minecraftforge/fmp/block/BlockMultipartContainer.java

```diff
@@ -0,0 +1,561 @@
+        TileMultipartContainer tile = getMultipartTile(world, pos);
+        if (tile != null)
+        {
+            tile.getPartContainer().randomDisplayTick(rand);
+        }
+    }
+
+    @Override
+    @SideOnly(Side.CLIENT)
+    public boolean addDestroyEffects(World world, BlockPos pos, ParticleManager particleManager)
+    {
+        RayTraceResult hit = reTrace(world, pos, ForgeMultipart.proxy.getPlayer());
+        if (hit != null)
+        {
+            if (hit.partHit.addDestroyEffects(AdvancedParticleManager.getInstance(particleManager))) { return true; }
+
+            ResourceLocation path = hit.partHit.getModelPath();
+            IBlockState state = hit.partHit.getExtendedState(MultipartRegistry.getDefaultState(hit.partHit).getBaseState());
+            IBakedModel model = path == null ? null
+                    : Minecraft.getMinecraft().getBlockRendererDispatcher().getBlockModelShapes().getModelManager().getModel(
+                            new ModelResourceLocation(path, MultipartStateMapper.instance.getPropertyString(state.getProperties())));
+            if (model != null)
+            {
+                TextureAtlasSprite icon = model.getParticleTexture();
+                if (icon != null)
+                {
+                    AdvancedParticleManager.getInstance(particleManager).addBlockDestroyEffects(pos, icon);
+                }
+            }
+        }
+        return true;
+    }
+
+    @Override
+    @SideOnly(Side.CLIENT)
+    public boolean addHitEffects(IBlockState state, World world, RayTraceResult target, ParticleManager particleManager)
+    {
+        RayTraceResult hit = target instanceof RayTraceResult ? (RayTraceResult) target : null;
+        if (hit != null)
+        {
+            if (hit.partHit.addHitEffects(hit, AdvancedParticleManager.getInstance(particleManager))) { return true; }
+
+            ResourceLocation path = hit.partHit.getModelPath();
+            IBlockState partState = hit.partHit.getExtendedState(MultipartRegistry.getDefaultState(hit.partHit).getBaseState());
+            IBakedModel model = path == null ? null
+                    : Minecraft.getMinecraft().getBlockRendererDispatcher().getBlockModelShapes().getModelManager().getModel(
+                            new ModelResourceLocation(path, MultipartStateMapper.instance.getPropertyString(partState.getProperties())));
+            if (model != null)
+            {
+                TextureAtlasSprite icon = model.getParticleTexture();
+                if (icon != null)
+                {
+                    AdvancedParticleManager.getInstance(particleManager).addBlockHitEffects(target.getBlockPos(), hit, bounds, icon);
+                }
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public boolean addLandingEffects(IBlockState state, WorldServer worldObj, BlockPos blockPosition, IBlockState iblockstate,
+            EntityLivingBase entity, int numberOfParticles)
+    {
+        return true;
```

Comment:
May be something to look into, but it could be just as simple as a raytrace of the player's center position straight down. But that's not important right now, just something to keep in mind.